File: app/src/main/java/com/rocky/moreapps/example/UpdaterExampleFragment.java
package com.rocky.moreapps.example;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;

import io.github.raghavsatyadev.moreapps.ForceUpdater;
import io.github.raghavsatyadev.moreapps.MoreAppsBuilder;
import io.github.raghavsatyadev.moreapps.MoreAppsLifecycleListener;
import io.github.raghavsatyadev.moreapps.model.MoreAppsDetails;
import io.github.raghavsatyadev.moreapps.utils.MoreAppsUtils;

public class UpdaterExampleFragment extends Fragment implements View.OnClickListener {

    public static UpdaterExampleFragment getInstance() {
        UpdaterExampleFragment fragment = new UpdaterExampleFragment();
        Bundle args = new Bundle();
        fragment.setArguments(args);
        return fragment;
    }

    @Override
    public View onCreateView(
            @NonNull LayoutInflater inflater, ViewGroup container,
            Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_updater_example, container, false);
        view.findViewById(R.id.btn_1).setOnClickListener(this);
        view.findViewById(R.id.btn_2).setOnClickListener(this);
        view.findViewById(R.id.btn_3).setOnClickListener(this);

        return view;
    }

    @Override
    public void onClick(View view) {
        int id = view.getId();
        if (id == R.id.btn_1) {
            option1();
        } else if (id == R.id.btn_2) {
            option2();
        } else if (id == R.id.btn_3) {
            option3();
        }
    }

    /**
     * call {@link MoreAppsBuilder#build()} first
     */
    public void option1() {
        if (ForceUpdater.shouldShowUpdateDialogs(getContext())) {
            ForceUpdater.showUpdateDialogs(getContext(), () -> {

            });
        }

//        call moreAppsDialog.removeUpdateListener(); in onStop() or onDestroy() or onDestroyView()
    }

    private void option2() {
        ForceUpdater.showDialogLive(getContext(),
                this,
                R.style.CustomDialog,
                new MoreAppsLifecycleListener() {
                    @Override
                    public void onStart() {
                    }

                    @Override
                    public void onStop() {
                    }

                    @Override
                    public void showingDialog() {
                    }

                    @Override
                    public void onComplete() {
                    }
                });
    }

    /**
     * call {@link MoreAppsBuilder#build()} first
     */
    public void option3() {
        MoreAppsDetails currentAppModel = MoreAppsUtils.getCurrentAppModel(getContext());
        switch (ForceUpdater.dialogToShow(getContext(), currentAppModel)) {
            case HARD_REDIRECT:
                ForceUpdater.showHardRedirectDialog(getContext(), currentAppModel, () -> {

                });
                break;
            case SOFT_REDIRECT:
                ForceUpdater.showSoftRedirectDialog(getContext(), currentAppModel, () -> {

                });
                break;
            case HARD_UPDATE:
                ForceUpdater.showHardUpdateDialog(getContext(), currentAppModel, () -> {
                });
                break;
            case SOFT_UPDATE:
                ForceUpdater.showSoftUpdateDialog(getContext(), currentAppModel, () -> {

                });
                break;
            case NONE:
//                    do nothing
                break;
        }
    }
}

File: app/src/main/java/com/rocky/moreapps/example/UpdaterExampleFragment.kt
package com.rocky.moreapps.example

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.View.OnClickListener
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import com.rocky.moreapps.example.R.layout
import com.rocky.moreapps.example.R.style
import io.github.raghavsatyadev.moreapps.ForceUpdater
import io.github.raghavsatyadev.moreapps.ForceUpdater.UpdateDialogType.HARD_REDIRECT
import io.github.raghavsatyadev.moreapps.ForceUpdater.UpdateDialogType.HARD_UPDATE
import io.github.raghavsatyadev.moreapps.ForceUpdater.UpdateDialogType.NONE
import io.github.raghavsatyadev.moreapps.ForceUpdater.UpdateDialogType.SOFT_REDIRECT
import io.github.raghavsatyadev.moreapps.ForceUpdater.UpdateDialogType.SOFT_UPDATE
import io.github.raghavsatyadev.moreapps.ForceUpdater.dialogToShow
import io.github.raghavsatyadev.moreapps.ForceUpdater.shouldShowUpdateDialogs
import io.github.raghavsatyadev.moreapps.ForceUpdater.showHardRedirectDialog
import io.github.raghavsatyadev.moreapps.ForceUpdater.showHardUpdateDialog
import io.github.raghavsatyadev.moreapps.ForceUpdater.showSoftRedirectDialog
import io.github.raghavsatyadev.moreapps.ForceUpdater.showSoftUpdateDialog
import io.github.raghavsatyadev.moreapps.MoreAppsLifecycleListener
import io.github.raghavsatyadev.moreapps.listener.MoreAppsUpdateDialogListener
import io.github.raghavsatyadev.moreapps.utils.MoreAppsUtils.getCurrentAppModel

class UpdaterExampleFragment : Fragment(), OnClickListener {
    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?,
    ): View? {
        val view = inflater.inflate(layout.fragment_updater_example, container, false)
        view.findViewById<View>(R.id.btn_updater_1).setOnClickListener(this)
        view.findViewById<View>(R.id.btn_updater_2).setOnClickListener(this)
        view.findViewById<View>(R.id.btn_updater_3).setOnClickListener(this)
        return view
    }

    override fun onClick(view: View) {
        when (view.id) {
            R.id.btn_updater_1 -> {
                option1()
            }

            R.id.btn_updater_2 -> {
                option2()
            }

            R.id.btn_updater_3 -> {
                option3()
            }
        }
    }

    /** call [io.github.raghavsatyadev.moreapps.MoreAppsBuilder.build] first */
    private fun option1() {
        if (shouldShowUpdateDialogs(requireContext())) {
            ForceUpdater.showUpdateDialogs(requireContext()) {
                // on close
            }
        }

//        call moreAppsDialog.removeUpdateListener(); in onStop() or onDestroy() or onDestroyView()
    }

    private fun option2() {
        ForceUpdater.showDialogLive(requireContext(),
            this,
            style.CustomDialog,
            object : MoreAppsLifecycleListener {
                override fun onStart() {
                    // Activity or Fragment's onStart LifeCycle Method
                }

                override fun onStop() {
                    // Activity or Fragment's onStop LifeCycle Method
                }

                override fun showingDialog() {
                    // stop other work till this dialog is showing
                }

                override fun onComplete() {
                    // do other work
                }
            })
    }

    /** call [io.github.raghavsatyadev.moreapps.MoreAppsBuilder.build] first */
    private fun option3() {
        val currentAppModel = getCurrentAppModel(requireContext())
        when (dialogToShow(requireContext(), currentAppModel)) {
            HARD_REDIRECT -> showHardRedirectDialog(requireContext(),
                currentAppModel,
                MoreAppsUpdateDialogListener {
                    // on close
                })

            SOFT_REDIRECT -> showSoftRedirectDialog(requireContext(),
                currentAppModel,
                MoreAppsUpdateDialogListener {
                    // on close
                })

            HARD_UPDATE -> showHardUpdateDialog(requireContext(),
                currentAppModel,
                MoreAppsUpdateDialogListener {
                    // on close
                })

            SOFT_UPDATE -> showSoftUpdateDialog(requireContext(),
                currentAppModel,
                MoreAppsUpdateDialogListener {
                    // on close
                })

            NONE -> {
                // nothing to update
            }
        }
    }

    companion object {
        val instance: UpdaterExampleFragment
            get() {
                val fragment = UpdaterExampleFragment()
                val args = Bundle()
                fragment.arguments = args
                return fragment
            }
    }
}